src/cargo/sources/registry/remote.rs

```diff
@@ -212,8 +221,18 @@ impl<'cfg> RegistryData for RemoteRegistry<'cfg> {
         let url = self.source_id.url();
         let refspec = "refs/heads/master:refs/remotes/origin/master";
         let repo = self.repo.borrow_mut().unwrap();
-        git::fetch(repo, url, refspec, self.config)
-            .chain_err(|| format!("failed to fetch `{}`", url))?;
+        let result = git::fetch(repo, url, refspec, self.config);
+        if let Err(e) = result {
+            let extra = if maybe_spurious(&e) && nightly_features_allowed() && !repo.is_empty()? {
+                "\nIf the network is unavailable, consider running with the -Zoffline \
+                 flag to run in offline mode."
```

Comment:
This warning I'm somewhat wary about because it feels a little aggressive about mentioning `-Zoffline`. We're already pretty fast-and-loose with what we interpret as spurious network error (we're rarely *sure* that it's a spurious error, it's just a random bucket of things that are all somewhat network related).

I'm personally wary of CI tools that make suggestions which often don't end up being applicable as well because it can be a frustrating experience. Ideally we'd recognize a spurious error, actually run resolution in offline mode, realize it works, and only then actually emit an appropriate error. That seems a bit heavy handed and a bit far fetched though.

Do you think though that if we don't have this sort of an error message then `-Zoffline` will be difficult to discover?

Reply:
Yea, this is only for discoverability.  Should I just remove this, or would it be worthwhile to emit a warning and see if it can succeed to resolve?  I lean towards just removing it.

Reply:
This also fires for `cargo install` which will give a bad suggestion, `cargo install foo -Z offline` will fail, so I think the risk of a bad suggestion is a little too high.

Reply:
Hm those are good points yeah. I'm fine with landing without this and we can sort of see over time whether we feel that we need to improve the error messages for offline.